tests/mocks/validators.go

```diff
@@ -0,0 +1,81 @@
+package mocks
```

Comment:
Usually the package for testing helpers/mocks should be named `testing` instead for consistency with other repositories

Reply:
I haven't seen this until now, especially since this package is intended for generation of testing data, not for testing itself, but ok

Reply:
Is your intention the package to be named as the go-testing package, also because the code is continuation of the language anyway?
In that way it could be like this:
coreth/tests/
 - validators_mocks.go
 - something_else_mocks.go
 etc.
OR
 - validators_utils.go

Reply:
See https://github.com/optakt/flow-dps/tree/master/testing

Reply:
WDYM? 
I see testing/mocks/mocks...?
What package name did you mean to change - the `tests` or `mocks`?

Reply:
There is just `testing/mocks` and `testing/helpers` in Flow  DPS. It should be the same here IMO if we wanted to have a common folder for mocks.

In this specific case, since the mocks are only used in the `plugin/evm` package though, I'm not convinced it's even needed to have them in a separate package now that I think about it, and anyway it's also not what your PR is doing since what you called mocks here is actually generic values used for testing. The mocks are defined in the `evm` package currently.

So what I would suggest is the following:

* If we need `RandID` and the function that generates test validator maps outside of this package, we create a package in `./testing/mocks` from the root of the repository with a file called `validators.go` which contains functions for generating validator maps and one named `short_ids.go` which contains the function currently named `RandID()`.
* Otherwise, we just hardcode the maps of validators that we need in the functions that need them. For example in my tests to have a fake validators map I just did this: https://github.com/flare-foundation/coreth/pull/33/files#diff-f74d73d6d9de90c767d318d06215fae9a4c4b74e3048fcc86043b6bb103131c3R49-R51 which I think is much simpler and clearer.

From what I remember, and correct me if I am wrong, the only place where we actually need a somewhat complex map of validators with multiple values is for the validators normalizer since we need to make sure the resulting map has the correct weight ratios right?

In which case, I think it would be best to just use my approach for all tests, and hardcode very simple maps everywhere, except in the relevant tests where we'd use a local test helper which generates a map with random values and expects a matching weight ratio map. WDYT?

Reply:
Okay, `testing/mocks` is what I see in Flow-dps of course, just asked because I wasn't sure if you meant the `tests` package or `mocks` one in the first comment and yes - it makes sense `tests` to be renamed to `testing`.

My intention for using the validators helper functions outside the `evm` was to have a single place for using tests generated values across the code, instead of multiple files and packages and sub-packages.
I also see that `ids.ShortID` is used in other places outside `evm` package - `peer` and `message`, maybe it will be needed when they are tested.

We have the generated map of `ids.ShortID:uint64` in validators_cache_test as well, and also since this map is used for inserting the cache in the type, it is also wrapped further in transitioner and manager types.

I am not sure what do you mean here - to remove all of my testing funcs from tests/mocks and simply use hardcoded maps?

Reply:
My suggestion is:

* the only func we need to have in the `testing/mocks` package IMO is the one to generate short IDs
* validator maps should be hardcoded in existing tests
	* except in the validator normalizer tests where we should have a function to generate a random validator map (with random IDs and random weights) but that function should remain in `validators_normalizer_test.go` and not be in a package meant to be used externally

Does that make sense?

Reply:
In terms of explanation yes.
`type ValidatorsTestData map[interface{}]map[ids.ShortID]uint64` is a test type that corresponds to the map type used by validators manager, transitioner, ftso and normalizer.
The function generating the validators tests data by epoch can also potentially be used in multiple places for testing.
So, if we leave only the function `RandID()` inside tests/mocks, I think the other testing data should still be placed in a separate helper/testing or whatever file or sub-package inside pluging/evm.

Also, it seems to me that this reorganization is not related to the PR anymore, what do you think of bringing it to another one?
Also it might be better to have a discussion outside the github comments and decide how to move forward first, then place the decided description in an issue and move to another PR solving it, what is your opinion about this?

Reply:
This PR is the one that adds this code to generate the validator maps, so IMO it should be solved in this PR before it gets merged 😅 If you really want to, we can merge this one as-is and open an issue about it to deal with it later, but I don't really see what is unclear/unsure at the moment 🤔 Maybe we should just get on a call to agree on a solution?

Reply:
I was speaking about the reorganization of the files only.
I thought it appeared as a question along with the work here, that is why I suggested it.
But of course when you have time, I would be happy to discuss it, and clear it out here also.